test/components/views/rooms/SearchBar-test.tsx

```diff
@@ -40,95 +35,65 @@ jest.mock("../../../../src/KeyBindingsManager", () => ({
         { getAccessibilityAction: jest.fn(() => mockCurrentEvent) })),
 }));
 
-/** mock out SearchWarning component so it doesn't affect the result of our test */
-jest.mock('../../../../src/components/views/elements/SearchWarning', () => ({
-    __esModule: true,
-    WarningKind: {
-        get Search() { // eslint-disable-line @typescript-eslint/naming-convention
-            return mockWarningKind;
-        },
-    },
-    default: jest.fn(({ children }) => (
-        <div>{ children }</div>
-    )),
-}));
-
-/** mock out PosthogTrackers component so it doesn't affect the result of our test */
-jest.mock('../../../../src/PosthogTrackers', () => ({
-    __esModule: true,
-    PosthogScreenTracker: jest.fn(({ children }) => (
-        <div>{ children }</div>
-    )),
-}));
-
 describe("SearchBar", () => {
-    beforeEach(() => {
-        wrapper = mount(<SearchBar {...searchProps} />);
-    });
-
     afterEach(() => {
-        wrapper.unmount();
         searchProps.onCancelClick.mockClear();
         searchProps.onSearch.mockClear();
     });
 
-    it("must render child components and pass necessary props", () => {
-        const postHogScreenTracker = wrapper.find(PosthogScreenTracker);
-        const searchWarning = wrapper.find(SearchWarning);
-
-        expect(postHogScreenTracker.length).toBe(1);
-        expect(searchWarning.length).toBe(1);
-        expect(postHogScreenTracker.props().screenName).toEqual("RoomSearch");
-        expect(searchWarning.props().isRoomEncrypted).toEqual(searchProps.isRoomEncrypted);
-        expect(searchWarning.props().kind).toEqual(mockWarningKind);
-    });
-
     it("must not search when input value is empty", () => {
-        const roomButtons = wrapper.find(".mx_SearchBar_button");
-        const searchButton = wrapper.find(".mx_SearchBar_searchButton");
+        const { container } = render(<SearchBar {...searchProps} />);
+        const roomButtons = container.querySelectorAll(".mx_SearchBar_button");
+        const searchButton = container.querySelectorAll(".mx_SearchBar_searchButton");
 
-        expect(roomButtons.length).toEqual(4);
+        expect(roomButtons.length).toEqual(2);
```

Comment:
Why did we lose buttons?

Reply:
See https://github.com/matrix-org/matrix-react-sdk/pull/9584#discussion_r1026164649